database/articles.js

```diff
@@ -1,42 +1,55 @@
 'use strict';
 
 const express = require('express');
-const expressApp = express();
-const bodyParser = require('body-parser');
-const jsonParser = bodyParser.json();
+const expressApp = express.Router();
 
-let articles = [];
+let articles = { articles: [] };
 
 const getAllArticles = () => {
   return articles;
 };
 
+const getSpecificArticle = (req) => {
+  for (let i = 0; i <= articles.articles.length; i++) {
+    if (`/${articles.articles[i].title}` === req) {
+      return articles.articles[i];
+    }
+  }
+};
+
 const postArticle = (req, res) => {
+  if (!req.body) res.send({ success: false });
+
+  articles.articles.push(req.body);
+  res.send({ success: true });
+};
+
+const putArticle = (req, res) => {
   if (req.body) {
-    articles.push(req.body);
+    for (let i = 0; i < articles.articles.length; i++) {
+      if (articles.articles[i].title === req) articles.articles[i] = req.body;
```

Comment:
This doesn't seem like this would work. `req` is an object and you're trying to see if it's equal to a string? Also, the body of an if should always be wrapped in { }